Untangle change history from redrafted

tuzz opened the pull request:
This change makes ChangeHistory a first-class
object that exposes some methods for adding and
updating items. This means the app doesn’t need
to know the details of how the change history is
represented.

Other than refactoring, the driver for this change
is to remove all reference of ‘redrafted’ from the
code base. Previously, change history was so
tangled up with ‘redrafted’ that it makes sense to
fix this first before continuing with that work.

tuzz committed:
Extract ChangeHistory as an object

This change makes ChangeHistory a first-class
object that exposes some methods for adding and
updating items. This means the app doesn’t need
to know the details of how the change history is
represented.

Other than refactoring, the driver for this change
is to remove all reference of ‘redrafted’ from the
code base. Previously, change history was so
tangled up with ‘redrafted’ that it makes sense to
fix this first before continuing with that work.

Seraphiana commented on spec/presenters/actions_presenter_spec.rb:
```diff
@@ -65,19 +65,13 @@
-
     context "when the document is redrafted and the update_type is minor" do
-      let(:payload) { FactoryGirl.create(:cma_case, publication_state: "redrafted", update_type: "minor") }
+      let(:payload) { FactoryGirl.create(:cma_case, first_published_at: "2016-01-01", publication_state: "redrafted", update_type: "minor") }
```
should this not be publication_state draft? I guess it doesn't need to change just yet, its just a little confusing in this weird still-sort-of-supporting-redrafted state

Seraphiana commented on spec/models/document_spec.rb:
```diff
@@ -689,30 +673,27 @@ def initialize(params = {})
 
-    it 'should not add note when no update type' do
-      document.update_type = ''
+    it "does not the change note when the update_type is not set" do
```
minor typo

Commit:
bug fix and new trait

fixed an issue where redrafted was an unrecognised state, and added
a new trait to the factories for redrafted documents to use
